### PR TITLE
New tests e2e BQ

### PR DESCRIPTION
**Pull request by priyabhatnagar25**

### CODE REVIEW

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 1]
```diff
@@ -924,4 +924,111 @@ public static void createSourceBQUpdateTable() throws IOException, InterruptedEx
   public static void createGcsBucketWithAvro() throws IOException, URISyntaxException {
     gcsSourceBucketName = createGCSBucketWithFile(PluginPropertyUtils.pluginProp("gcsAvroAllDataFile"));
   }
+
+  @Before(order = 1, value = "@BQ_UPSERT_SOURCE_TEST")
```
Comment (itsmekumari) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:928]: Add the after hooks for each of these before hooks, to cleanup the created tables after the scenario execution.

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 2]
```diff
@@ -924,4 +924,111 @@ public static void createSourceBQUpdateTable() throws IOException, InterruptedEx
...
+    bqSourceTable = "E2E_SOURCE_" + UUID.randomUUID().toString().replaceAll("-" , "_");
+    io.cdap.e2e.utils.BigQueryClient.getSoleQueryResult("create table `" + datasetName + "." + bqSourceTable + "` " +
+                                                          "(ID INT64, Name STRING, " + "Price FLOAT64," +
+                                                          "Customer_Exists BOOL ) ");
+    try {
+      io.cdap.e2e.utils.BigQueryClient.getSoleQueryResult("INSERT INTO `" + datasetName + "." + bqSourceTable + "` " +
+                                                            "(ID,  Name, Price, Customer_Exists)" +
+                                                            "VALUES" +  "(5, 'Raja', 500.0, true)," +
+                                                            "(6, 'Tom', 100.0, false)");
+
```
Comment (itsmekumari) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: Remove extra lines

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 3]
```diff
@@ -924,4 +924,111 @@ public static void createSourceBQUpdateTable() throws IOException, InterruptedEx
...
+    } catch (NoSuchElementException e) {
+      // Insert query does not return any record.
+      // Iterator on TableResult values in getSoleQueryResult method throws NoSuchElementException
+    }
+    PluginPropertyUtils.addPluginProp("bqSourceTable", bqSourceTable);
+    BeforeActions.scenario.write("BQ Source Table " + bqSourceTable + " created successfully");
+  }
+
+
```
Comment (itsmekumari) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: Remove extra lines

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 4]
```diff
@@ -924,4 +924,111 @@ public static void createSourceBQUpdateTable() throws IOException, InterruptedEx
...
+    BeforeActions.scenario.write("BQ target table name - " + bqTargetTable);
+    io.cdap.e2e.utils.BigQueryClient.getSoleQueryResult("create table `" + datasetName + "." + bqTargetTable + "` " +
+                                                          "(ID INT64, Name STRING, " + "Price FLOAT64," +
+                                                          "Customer_Exists BOOL ) ");
+    try {
+      io.cdap.e2e.utils.BigQueryClient.getSoleQueryResult("INSERT INTO `" + datasetName + "." + bqTargetTable + "` " +
+                                                            "(ID,  Name, Price, Customer_Exists)" +
+                                                            "VALUES" + "(5, 'Rakesh', 500.0, true)");
+
+
```
Comment (itsmekumari) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: same here

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 5]
```diff
@@ -924,4 +924,111 @@ public static void createSourceBQUpdateTable() throws IOException, InterruptedEx
...
+    } catch (NoSuchElementException e) {
+      // Insert query does not return any record.
+      // Iterator on TableResult values in getSoleQueryResult method throws NoSuchElementException
+    }
+    PluginPropertyUtils.addPluginProp("bqTargetTable", bqTargetTable);
+    BeforeActions.scenario.write("BQ Target Table " + bqTargetTable + " created successfully");
+  }
```
Comment (itsmekumari) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:965]: Add a line

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 6]
```diff
@@ -924,4 +924,111 @@ public static void createSourceBQUpdateTable() throws IOException, InterruptedEx
...
+                                          "(Address,  id, Firstname, LastName)" +
+                                          "VALUES" +  "('Agra', 1, 'Harry','')," +
+                                          "('Noida', 2, '','')");
+    } catch (NoSuchElementException e) {
+      // Insert query does not return any record.
+      // Iterator on TableResult values in getSoleQueryResult method throws NoSuchElementException
+    }
+    PluginPropertyUtils.addPluginProp(" bqSourceTable",  bqSourceTable);
+    BeforeActions.scenario.write("BQ Source Table " +  bqSourceTable + " updated successfully");
+  }
```
Comment (itsmekumari) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:986]: Add a line

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 7]
```diff
@@ -924,4 +924,111 @@ public static void createSourceBQUpdateTable() throws IOException, InterruptedEx
...
+                                                            "VALUES" +  "('string_1', 1, 0.1,true)," +
+                                                            "('string_1', 2, 0.2,false)");
+    } catch (NoSuchElementException e) {
+      // Insert query does not return any record.
+      // Iterator on TableResult values in getSoleQueryResult method throws NoSuchElementException
+    }
+    PluginPropertyUtils.addPluginProp(" bqSourceTable",  bqSourceTable);
+    BeforeActions.scenario.write("BQ Source Table " +  bqSourceTable + " updated successfully");
+  }
+  @Before(value = "@BQ_UPDATE_SINK_DEDUPE_TEST")
```
Comment (itsmekumari) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:1009]: same here

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 8]
```diff
@@ -924,4 +924,111 @@ public static void createSourceBQUpdateTable() throws IOException, InterruptedEx
...
+  public static void createSinkBQDedupeTable() throws IOException, InterruptedException {
+
```
Comment (itsmekumari) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: Remove extra lines, re-check the file for same.

File: src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature [hunk 9]
```diff
@@ -195,3 +195,141 @@ Feature: BigQuery sink - Verification of BigQuery to BigQuery successful data tr
     Then Open and capture logs
     Then Verify the pipeline status is "Succeeded"
     Then Verify the partition table is created with partitioned on field "bqPartitionFieldTime"
+
+  @BQ_UPSERT_SOURCE_TEST @BQ_UPSERT_SINK_TEST @EXISTING_BQ_CONNECTION
+  Scenario:Validate successful records transfer from BigQuery to BigQuery with Advanced operations Upsert with updating the destination table schema and destination table must exist.
```
Comment (itsmekumari) [inline src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature]: Change to --- Validate successful records transfer from BigQuery source to BigQuery sink with Upsert operation by updating destination table schema and destination table exists with records in it.

File: src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature [hunk 10]
```diff
@@ -195,3 +195,141 @@ Feature: BigQuery sink - Verification of BigQuery to BigQuery successful data tr
...
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Close the pipeline logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Validate the data transferred from BigQuery to BigQuery with actual And expected file for: "bqUpsertExpectedFile"
+
+  @BQ_NULL_MODE_SOURCE_TEST @BQ_SINK_TEST @EXISTING_BQ_CONNECTION
+  Scenario: Validate Successful record  BigQuery source plugin with all NULL values in one column and Few NULL value in different cloumn.
```
Comment (itsmekumari) [inline src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature]: Change ---- Validate Successful records transfer from BigQuery source plugin with all NULL values in one column and Few NULL values in different column.

File: src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature [hunk 11]
```diff
@@ -195,3 +195,141 @@ Feature: BigQuery sink - Verification of BigQuery to BigQuery successful data tr
...
+    Then Click on the Get Schema button
+    Then Validate "BigQuery" plugin properties
+    And Close the Plugin Properties page
+    Then Navigate to the properties page of plugin: "BigQuery2"
+    Then Replace input plugin property: "project" with value: "projectId"
+    Then Enter input plugin property: "datasetProject" with value: "projectId"
+    Then Enter input plugin property: "referenceName" with value: "BQReferenceName"
+    Then Enter input plugin property: "dataset" with value: "dataset"
+    Then Enter input plugin property: "table" with value: "bqTargetTable"
+    Then Click plugin property: "updateTableSchema"
```
Comment (itsmekumari) [inline src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature]: Please check with ITN class as update table schema is false

File: src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature [hunk 12]
```diff
@@ -195,3 +195,141 @@ Feature: BigQuery sink - Verification of BigQuery to BigQuery successful data tr
...
+    Then Wait till pipeline preview is in running state
+    Then Open and capture pipeline preview logs
+    Then Verify the preview run status of pipeline in the logs is "succeeded"
+    Then Close the pipeline logs
+    Then Close the preview
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
```
Comment (itsmekumari) [inline src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature:451]: Validation step to be added

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 1 above)
Comment (priyabhatnagar25) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:928]: resolved

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 2 above)
Comment (priyabhatnagar25) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: removed

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 3 above)
Comment (priyabhatnagar25) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: resolved

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 4 above)
Comment (priyabhatnagar25) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: removed

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 5 above)
Comment (priyabhatnagar25) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:965]: added

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 6 above)
Comment (priyabhatnagar25) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:986]: added

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 7 above)
Comment (priyabhatnagar25) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:1009]: added

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 8 above)
Comment (priyabhatnagar25) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: removed

File: src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature (hunk 9 above)
Comment (priyabhatnagar25) [reply src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature]: changed

File: src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature (hunk 10 above)
Comment (priyabhatnagar25) [reply src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature]: changed

File: src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature (hunk 11 above)
Comment (priyabhatnagar25) [reply src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature]: removed the step

File: src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature (hunk 12 above)
Comment (priyabhatnagar25) [reply src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature:451]: added validation step

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 13]
```diff
@@ -269,7 +269,9 @@ public static void createTempSourceBQTable() throws IOException, InterruptedExce
   }
 
   @After(order = 1, value = "@BQ_SOURCE_TEST or @BQ_PARTITIONED_SOURCE_TEST or @BQ_SOURCE_DATATYPE_TEST or " +
-    "@BQ_INSERT_SOURCE_TEST or @BQ_UPDATE_SINK_TEST")
+    "@BQ_INSERT_SOURCE_TEST or @BQ_UPDATE_SINK_TEST" + "@BQ_TIME_STAMP_SOURCE_TEST or @BQ_INSERT_INT_SOURCE_TEST" +
```
Comment (itsmekumari) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: Please add the after hooks properly.

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 14]
```diff
@@ -917,11 +919,182 @@ public static void createSourceBQUpdateTable() throws IOException, InterruptedEx
...
+      // Iterator on TableResult values in getSoleQueryResult method throws NoSuchElementException
+    }
+    PluginPropertyUtils.addPluginProp(" bqSourceTable", bqSourceTable);
+    BeforeActions.scenario.write("BQ Source Table " + bqSourceTable + " updated successfully");
+  }
+
+  @Before(value = "@BQ_UPDATE_SOURCE_DEDUPE_TEST")
+  public static void createSourceBQDedupeTable() throws IOException, InterruptedException {
+
+
```
Comment (itsmekumari) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: Remove extra lines

File: src/e2e-test/resources/pluginParameters.properties [hunk 15]
```diff
@@ -191,6 +192,18 @@ bqInvalidRefName=invalidRef&^*&&*
...
+bqUpdatededupeExpectedFile=testdata/BigQuery/BQUpdateDedupeFile
+bqInsertExpectedFile=testdata/BigQuery/BQInsertIntFile
+relationTableKeyValue=Name
+dedupeByOrder=ASC
+dedupeByValue=ID
+rangeStartValue=2
+rangeEndValue=3
+rangeIntervalValue=1
+partitionByFieldValue=ID
+partiontion_by_field_value
```
Comment (itsmekumari) [inline src/e2e-test/resources/pluginParameters.properties]: Are we not passing any value here?

**Pull request closed without merging**